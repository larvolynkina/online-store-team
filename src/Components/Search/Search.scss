.search {
    position: relative;
    width: 100%;

    &__input {
        box-sizing: border-box;
        width: 100%;
        padding: 10px 40px;
        background: #FFFFFF;
        box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.135216);
        border-radius: 8px;
        border: none;
        font-weight: 500;
        font-size: 14px;
        color: #999999;
        outline: none;
        @media screen and (max-width: 730px) {
            padding: 7px 40px;
        }
        &:focus {
            box-shadow: inset 0px 0px 5px 1px #6f64f8;
        }
    }

    &__clearBtn {
        position: absolute;
        background-color: transparent;
        background-image: url('./../../assets/icons/cancel.svg');
        background-size: 0%;
        right: 10px;
        top: 9px;
        width: 18px;
        height: 18px;
        opacity: 0.5;
        cursor: pointer;
        &_active {
            background-size: 100%;
        }
    }

    &::before {
        position: absolute;
        content: url('./../../assets/icons/search.svg');
        left: 10px;
        top: 8px;
        opacity: 0.5;
        @media screen and (max-width: 730px) {
            top: 5px;
        }
    }
}