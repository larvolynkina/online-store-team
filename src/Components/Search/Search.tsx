import React, { useState, useEffect } from 'react';
import { useSearchParams } from 'react-router-dom';
import './Search.scss';

function Search() {
  const [className, setClassName] = useState('search__clearBtn');
  const [searchParams, setSearchParams] = useSearchParams();
  const [searchValue, setSearchValue] = useState('');

  function clearSearchField(): void {
    setClassName('search__clearBtn');
    searchParams.delete('search');
    setSearchParams(searchParams);
  }

  function addSearchFilter(event: React.ChangeEvent<HTMLInputElement>): void {
    setSearchValue(event.target.value);
    if (event.target.value !== '') {
      setClassName('search__clearBtn search__clearBtn_active');
      searchParams.set('search', event.target.value);
      setSearchParams(searchParams);
    } else {
      clearSearchField();
    }
  }

  function isSearchFilter() {
    const search: string | null = searchParams.get('search');
    if (search) {
      setSearchValue(search);
      setClassName('search__clearBtn search__clearBtn_active');
    } else {
      setSearchValue('');
    }
  }

  useEffect(() => {
    isSearchFilter();
  }, [searchParams]);

  return (
    <div className="search">
      <input
        className="search__input"
        autoComplete="off"
        type="text"
        id="search"
        value={searchValue}
        placeholder="Search..."
        onChange={
          (event: React.ChangeEvent<HTMLInputElement>) => addSearchFilter(event)
        }
      />
      <button onClick={clearSearchField} className={className} type="button" aria-label="Clear search field" />
    </div>
  );
}

export default Search;
