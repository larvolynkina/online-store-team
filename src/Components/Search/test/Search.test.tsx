import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import Search from '../Search';

describe('search component', () => {
  it('renders search input', () => {
    render(<Search />, { wrapper: MemoryRouter });
    const searchInput: HTMLInputElement = screen.getByPlaceholderText(/Search.../i);
    expect(searchInput).toBeInTheDocument();
  });

  it('check search value and clear button', () => {
    render(<Search />, { wrapper: MemoryRouter });
    const searchInput: HTMLInputElement = screen.getByPlaceholderText(/Search.../i);
    expect(searchInput.value).toBe('');
    const clearButton = screen.getByLabelText('Clear search field');
    expect(clearButton).toBeInTheDocument();
    expect(clearButton).not.toHaveClass('search__clearBtn_active');
    userEvent.type(searchInput, 'apple');
    expect(searchInput.value).toBe('apple');
    expect(clearButton).toHaveClass('search__clearBtn_active');
    userEvent.click(clearButton);
    expect(searchInput.value).toBe('');
    expect(clearButton).not.toHaveClass('search__clearBtn_active');
  });

  it('search snapshot', () => {
    const search = render(<Search />, { wrapper: MemoryRouter });
    expect(search).toMatchSnapshot();
  });
});
