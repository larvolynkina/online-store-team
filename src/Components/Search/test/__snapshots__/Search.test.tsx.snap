// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search component search snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="search"
      >
        <input
          autocomplete="off"
          class="search__input"
          id="search"
          placeholder="Search..."
          type="text"
          value=""
        />
        <button
          aria-label="Clear search field"
          class="search__clearBtn"
          type="button"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="search"
    >
      <input
        autocomplete="off"
        class="search__input"
        id="search"
        placeholder="Search..."
        type="text"
        value=""
      />
      <button
        aria-label="Clear search field"
        class="search__clearBtn"
        type="button"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
